### PR TITLE
Added CLI commands

### PR DESCRIPTION
**Pull request by PaulCombal**

### CODE REVIEW

**Comment by PaulCombal** [comment]
What's up #43 gang

File: src/cli/cli_funcs.cpp [hunk 1]
```diff
@@ -1,45 +1,182 @@
...
+			return cli;
+		}
+
+		g_steam->launch_app(app);
+		g_steam->refresh_achievements();
+		auto achievements = g_steam->get_achievements();
+		g_steam->quit_game();
+		
+		// We want to order achievements from most unlocked to least unlocked
+		std::sort(achievements.begin(), achievements.end(), compareByUnlockRateDesc);
```
Comment (wgpierce) [inline src/cli/cli_funcs.cpp]: I'm wary of changing the ordering; it's differ from the GUI. Can it be an optional to sort them?

File: src/cli/cli_funcs.cpp [hunk 2]
```diff
@@ -1,45 +1,182 @@
...
+		std::cout << "API Name \t\tName \t\tDescription \t\tUnlock rate \t\tUnlocked" << std::endl;
+		std::cout << "--------------------------------------------------------------" << std::endl;
+		for ( Achievement_t& it : achievements )
+		{
+			std::cout 
+				<< it.id << " \t" 
+				<< it.name << " \t" 
+				<< it.desc << " \t" 
+				<< it.global_achieved_rate << "% \t"
+				<< (it.achieved ? "✔️" : "❌") << std::endl;
```
Comment (wgpierce) [inline src/cli/cli_funcs.cpp:133]: I'm not sure how well unicode will do on a commandline, but I guess it's fine

File: README.MD [hunk 3]
```diff
@@ -49,12 +49,24 @@ Run SamRewritten by opening the corresponding desktop entry, or by using the `sa
...
-We have big ambitions for SamRewritten, and I believe many of them are related to these command-line options, so stay put, because that list will be growing soon!
-
-* -a \<appid\>
-	* Prevents the window from showing and sets your status to "In Game"
-	* `./bin/launch.sh -a 10` will idle Counter Strike
+I know you linux geeks love to use command line options! SamRewritten has a few, let us know if you want to see more!
+
+```
+Usage:
+  samrewritten AppId [OPTION...]
```
Comment (wgpierce) [inline README.MD]: `AppId` can be `[AppId]` since it's optional.

File: src/cli/cli_funcs.cpp [hunk 4]
```diff
@@ -1,45 +1,182 @@
...
+void idle_app(AppId_t appid) 
+{
 	std::cout << "Idling from command line " << appid << std::endl;
-	AppId_t realid = strtoul(appid.c_str(), NULL, 10);
-	g_steam->launch_game(realid);
+	g_steam->launch_app(appid);
 	signal(SIGINT, handle_sigint_cli);
+
+	// Wait for ctrl+c, so we can kill both processes, otherwise
+    // GUI process will exit while game process still goes on
```
Comment (wgpierce) [inline src/cli/cli_funcs.cpp]: nit: comment misaligned

**Review by wgpierce — COMMENTED** [review]
looking good

File: src/cli/cli_funcs.cpp (hunk 1 above)
Comment (PaulCombal) [reply src/cli/cli_funcs.cpp]: I'll add that to my TODO list

File: README.MD (hunk 3 above)
Comment (PaulCombal) [reply README.MD]: Good catch

**Comment by PaulCombal** [comment]
@wgpierce Is everything ok now?

File: src/cli/cli_funcs.cpp [hunk 5]
```diff
@@ -107,8 +108,16 @@ bool go_cli_mode(int argc, char* argv[]) {
...
+
+			if (sort == "unlock_rate")
+			{
+				std::sort(achievements.begin(), achievements.end(), compareByUnlockRateDesc);
+			}
+		}
+		
 
 		// https://github.com/haarcuba/cpp-text-table -> worth? nah but best I've found
 		std::cout << "Sorted by global unlock rate\n" << std::endl;
```
Comment (wgpierce) [inline src/cli/cli_funcs.cpp]: now this isn't always true

**Review by wgpierce — COMMENTED** [review]
looks good after one more fix, no need to re-review

File: src/cli/cli_funcs.cpp (hunk 5 above)
Comment (PaulCombal) [reply src/cli/cli_funcs.cpp]: True thanks, I'll merge soon!

**Pull request merged**